File: Python/p3.py
N = float(input("N = "))
M = float(input("M = "))
sum = 0
i = 1
while True:
    prod = 1
    j = 0

    while j <= M:
        prod *= (2 * j + M) / i
        j += 1

    sum += (i + 3) / (i + 4) + prod
    i += 1

    if i > N:
        break
y = sum
print("y =", y)

File: Java/Practice/p3.java
package Practice;

import java.util.Scanner;

public class p3 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("N = ");
        int N = scanner.nextInt();
        System.out.print("M = ");
        int M = scanner.nextInt();
        double sum = 0;
        for (int i = 1; i <= N; i++) {
            double prod = 1.0;
            for (int j = 0; j <= M; j++) {
                prod *= (double) (2 * j + M) / i;
            }
            sum += (double) (i + 3) / (i + 4) + prod;
        }
        double y = sum;
        System.out.println("y = " + y);

    }
}
